packages/query-core/src/notifyManager.ts

```diff
@@ -10,7 +12,8 @@ type BatchCallsCallback<T extends Array<unknown>> = (...args: T) => void
 
 type ScheduleFunction = (callback: () => void) => void
 
-export const defaultScheduler: ScheduleFunction = (cb) => setTimeout(cb, 0)
+export const defaultScheduler: ScheduleFunction = (cb) =>
```

Comment:
note: we want to change this to `scheduleMicroTask` in the next major version, so maybe we should keep this outside of the managed timeouts to avoid larger changes when someone opts into their own managed timeout

Reply:
ah, I’m just now realizing that `systemSetTimeoutZero` doesn’t use the `timeoutManager` 😅 . I get it now 👍

Reply:
> so you’re saying places where we now use `systemSetTimeoutZero` we should in the future use `scheduleMicroTask` ?

I can't make that assessment - it's up to you to decide that should remain `globalThis.setTimeout(cb, 0)` and what should change to your new `scheduleMicroTask` function. 

But yeah, my intention is that changes from `setTimeout(cb, 0)` to `systemSetTimeoutZero` are syntactic only, and remain semantically unchanged compared to before this PR.